server/src/main/java/org/elasticsearch/common/network/InetAddresses.java

```diff
@@ -60,6 +64,12 @@ public static boolean isInetAddress(String ipString) {
                     return null;
                 }
             }
+            if (percentIndex == ipString.length() - 1) {
+                return null;
```

Comment:
Let's add a comment here that explains that we are doing this check to filter out strings that end in `%` and have an empty scope ID.